app-editors/jedit/jedit-5.4.0-r1.ebuild

```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+JAVA_PKG_IUSE="doc test"
+inherit desktop java-pkg-2 java-ant-2 xdg-utils
+
+DESCRIPTION="Programmer's editor written in Java"
+HOMEPAGE="http://www.jedit.org"
+SRC_URI="mirror://sourceforge/${PN}/${PN}${PV}source.tar.bz2"
+
+LICENSE="BSD GPL-2"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+SLOT="0"
+
+CP_DEPEND="dev-java/jsr305:0"
+
+RDEPEND="
+	${CP_DEPEND}
+	virtual/jre:1.8"
+
+DEPEND="
+	${CP_DEPEND}
+	virtual/jdk:1.8
+	dev-java/ant-contrib:0
+	dev-java/ant-apache-bsf:0
+	dev-java/bsh:0
+	test? (
+		dev-java/ant-junit:0
+		dev-java/hamcrest-library:1.3
+	)"
```

Comment:
please comment on why we need the java version restriction

Reply:
Done